core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -92,12 +95,34 @@ object ExternalWriteConverterCtx extends ExternalBackendUtils {
     uploadPlan
   }
 
-  def internalJavaClassOf(dt: DataType) : Class[_] = {
+  def internalJavaClassOf(dt: DataType): Class[_] = {
     dt match {
       case n if n.isInstanceOf[DecimalType] & n.getClass.getSuperclass != classOf[DecimalType] => Double.javaClass
-      case _ : DateType => Long.javaClass
-      case _ : UserDefinedType[_/*ml.linalg.Vector */] => classOf[Vector]
-      case _ : DataType => ReflectionUtils.supportedTypeOf(dt).javaClass
+      case _: DateType => Long.javaClass
+      case _: UserDefinedType[_ /*ml.linalg.Vector */ ] => classOf[Vector]
+      case _: DataType => ReflectionUtils.supportedTypeOf(dt).javaClass
     }
   }
+
+  def initFrame(key: String, columns: Array[String], timeout: Int): Unit = {
+    val (ip, port, timestamp) = leaderInfo()
+    val client = ExternalFrameWriterClient.create(ip, port, timestamp, timeout)
```

Comment:
Isn't there any overhead with establishing a connection to a leader. Shouldn't we create a connection to a leader only once?

Reply:
Good point!